labcontrol/gui/js_tests/addPlateModal_tests.js

```diff
@@ -0,0 +1,154 @@
+// Requires
+// QUnit
+
+// Library under test:
+// addPlateModal.js
+
+// Tests of function insertPlateModalDiv
+QUnit.module("insertPlateModalDiv", function (hooks) {
+    // Verify correct behavior using default arguments
+    QUnit.test("default arguments", function (assert) {
+        //jquery selector strings
+        let modalIdDivSelectorStr = "body div#addPlateModal";
+        let tableIdSelectorStr = "body table#searchPlateTable";
+
+        // before running function under test, body has no div with
+        // default modal id and no table with default table name
+        assert.equal($(modalIdDivSelectorStr).length, 0);
+        assert.equal($(tableIdSelectorStr).length, 0);
+
+        insertPlateModalDiv();
+
+        // after running function under test, table has these elements
+        assert.equal($(modalIdDivSelectorStr).length, 1);
+        assert.equal($(tableIdSelectorStr).length, 1);
+
+        // NOTE: unlike python unit tests, qunit does not abort the whole
+        // test case when an assert fails.  Therefore, the cleanup can
+        // simply be done in-line at the end of the test.
+        // Also note that this clean-up can't be done automatically as the dom
+        // objects were added (intentionally) to body rather than to the
+        // qunit-fixture div.
+        $(modalIdDivSelectorStr).remove();
+    });
+
+    // Verify correct behavior when arguments are explicitly specified
+    QUnit.test("explicit arguments", function (assert) {
+        //explicit names
```

Comment:
some comments begin immediately after, while some have a space after '//'. Not a personal concern, but in case there's a standard...